src/ecdsapod.rs

```diff
@@ -0,0 +1,546 @@
+    let st_type = type_statement();
+
+    // hash the msg as in-circuit
+    let msg_limbs = scalar_to_limbs(msg);
+    let msg_hash = PoseidonHash::hash_no_pad(&msg_limbs);
+
+    let st_msg = Statement::ValueOf(
+        AnchoredKey::from((SELF, KEY_SIGNED_MSG)),
+        Value::from(RawValue(msg_hash.elements)),
+    );
+
+    // hash the pk as in-circuit
+    let pk_x_limbs = base_to_limbs(pk.0.x);
+    let pk_y_limbs = base_to_limbs(pk.0.y);
+    let pk_hash = PoseidonHash::hash_no_pad(&[pk_x_limbs, pk_y_limbs].concat());
+
+    let st_pk = Statement::ValueOf(
+        AnchoredKey::from((SELF, KEY_ECDSA_PK)),
+        Value::from(RawValue(pk_hash.elements)),
+    );
+
+    vec![st_type, st_msg, st_pk]
+}
+
+fn scalar_to_limbs(v: Secp256K1Scalar) -> Vec<F> {
+    let max_num_limbs = 8;
+    let v_biguint = biguint_from_array(std::array::from_fn(|i| v.0[i]));
+    let mut limbs = v_biguint.to_u32_digits();
+    assert!(max_num_limbs >= limbs.len());
+    limbs.resize(max_num_limbs, 0);
+    let limbs_f: Vec<F> = limbs.iter().map(|l| F::from_canonical_u32(*l)).collect();
+    limbs_f
+}
+
+fn base_to_limbs(v: Secp256K1Base) -> Vec<F> {
+    let max_num_limbs = 8;
+    let v_biguint = biguint_from_array(std::array::from_fn(|i| v.0[i]));
+    let mut limbs = v_biguint.to_u32_digits();
+    assert!(max_num_limbs >= limbs.len());
+    limbs.resize(max_num_limbs, 0);
+    let limbs_f: Vec<F> = limbs.iter().map(|l| F::from_canonical_u32(*l)).collect();
+    limbs_f
+}
+
+fn biguint_from_array(arr: [u64; 4]) -> BigUint {
+    BigUint::from_slice(&[
+        arr[0] as u32,
+        (arr[0] >> 32) as u32,
+        arr[1] as u32,
+        (arr[1] >> 32) as u32,
+        arr[2] as u32,
+        (arr[2] >> 32) as u32,
+        arr[3] as u32,
+        (arr[3] >> 32) as u32,
+    ])
+}
+
+#[cfg(test)]
+pub mod tests {
+    use std::any::Any;
+
+    use plonky2::field::types::Sample;
+    use plonky2_ecdsa::curve::{
+        curve_types::{Curve, CurveScalar},
+        ecdsa::{ECDSAPublicKey, ECDSASecretKey, ECDSASignature, sign_message},
+        secp256k1::Secp256K1,
+    };
+    use pod2::{self, frontend::MainPodBuilder, op};
+
+    use super::*;
+
+    #[test]
+    fn test_ecdsa_pod_verify() -> Result<()> {
+        // first generate all the circuits data so that it does not need to be
+        // computed at further stages of the test (affecting the time reports)
+        timed!(
+            "generate ECDSA_VERIFY, STANDARD_ECDSA_POD_DATA, STANDARD_REC_MAIN_POD_CIRCUIT",
+            {
+                let (_, _) = &*ECDSA_VERIFY_DATA;
+                let (_, _) = &*STANDARD_ECDSA_POD_DATA;
+                let _ = &*pod2::backends::plonky2::STANDARD_REC_MAIN_POD_CIRCUIT_DATA;
+            }
+        );
+
+        let params = Params {
+            max_input_signed_pods: 0,
+            ..Default::default()
+        };
+
+        let sk = ECDSASecretKey::<Secp256K1>(Secp256K1Scalar::rand());
+        let msg = Secp256K1Scalar::rand();
```

Comment:
I would like to suggest using deterministic values here so that the tests are more reproducible.

For example:
```
let sk = ECDSASecretKey::<Secp256K1>(Secp256K1Scalar([123, 456, 789, 123]));
let msg = Secp256K1Scalar([321, 654, 987, 321]);
```

Reply:
cool, updated